### ISSUE TITLE
Downloading multiple files at once is not working on iOS

### ISSUE THREAD

**Issue by tanuva** [photoprism/photoprism#1437]
As it turned out in #1436, offering multiple files for downloading at the same time is not handled well by iOS. As of now, an "open with" screen shows up for the first file only.

There's no path to the next file from that screen, neither does it support handling multiple files (afaik). The only choices are to save/open the file in an app or to return via the browser's "back" button.

**Comment by tanuva** [comment]
A workaround that I came up with is to have a file selection screen that opens when tapping the download button. That way would also work fine-ish for deciding between original vs. transcoded video, assuming the user can be expected to make sense of that difference...

**Comment by swissmanu** [comment]
I cannot reproduce this. When I select multiple photos and then use the "Download" action from the lower right corners menu, I can download a ZIP file containing all photos. Using the "Files" app, I then can unzip this file and access all the photos individually.

I use iOS 14.7.1.

Might this issue be obsolete by the time?

**Comment by graciousgrey** [comment]
Tested it again, works for me on iOS 14.7.1 within the PWA and from Safari :) 
@tanuva could you test again as well ?

**Comment by tanuva** [comment]
The issue I saw originally was with stacked images (or rather: a video with a preview image that PP models as stacked images). PP offered downloads for all stacked images separately and at the same time, not as a zip archive. That didn't work for iOS's share sheet. In a desktop browser one would get multiple running downloads in parallel.

I suspect the workaround applied in #1436 (as mentioned in the original post) might have rendered this issue invalid. Tbh I don't fully understand how the turn from "it works now" to "it doesn't work yet?" in the last posts of that issue happened.

So, concluding: The case I intended to report here is still valid but not currently reproducible the way I noticed it.

**Comment by graciousgrey** [comment]
@tanuva sorry for the confusion. 
I wasn't aware that this ticket is about downloading stacks. 

In general downloading multiple files at once is working on iOS when those are ziped (e.g. when selecting multiple images and download those). You are right,  this is not the case for stacks. 

The ideal solution would be  #449  in combination with always provide multiple downloads as zip instead of parallel single downloads.

**Comment by lastzero** [comment]
@tanuva Is this still not working as expected?

**Comment by tanuva** [comment]
I pulled the latest stable build yesterday and tried downloading stacked + live photos with that. Downloading
* a live photo will offer the video for download (I'd expect the still picture)
* one stacked photo downloads the secondary file instead of the  primary one. (The names are roughly "image 1.jpg" (primary) and "image.jpg" (secondary).) Another one gives me the primary one.

**Comment by lastzero** [comment]
@tanuva Thanks for testing! It sounds like this could be a general issue, i.e. we might need to change (or configure) the priority with which files are downloaded (image > video, primary image > other images...)?

Have you played with these settings to see if it makes any difference?

- https://docs.photoprism.app/getting-started/config-files/settings/#downloads
- https://docs.photoprism.app/getting-started/config-files/settings/#albums

**Comment by tanuva** [comment]
I agree on the priority topic. There's a difference in photographer intent between videos and live photos that affects downloads:
* for live photos, I expect the photo to be downloaded
* for videos, I expect the video (instead of the still preview, for example)

Afaics, Photoprism can tell which short video is actually a live photo, so that might be doable.

I have to admit I haven't touched the linked settings yet, but looking at the documentation I at least don't see why they would help. I deal with `originals` only, so switching the file name for imported files won't have an effect. I don't have RAWs in PP (they stay in Capture One...), nor do I want the XMPs.

Are stacks internally the same as albums? Then the ordering might *change* something, but since I saw both primary and non-primary photos being downloaded, the behaviour might just get inverted. 🤔

Comparing two stacks where I get the primary photo for one and the secondary for the other, it almost seems like alphabetic ordering of the file name decides which one is offered (primary first):
* `IMG_3031.jpg` vs. `RNI-Flashback-IMG-AF0F776F-8E44-457C-A672-57D1F084A78F.JPG`: Downloads the primary
* `24-04-13_14-15-16_TNU01274 1.jpg` vs. `24-04-13_14-15-16_TNU01274.jpg`: Downloads secondary
